Catalyst support for crafted rare items

dclamage opened the pull request:
When crafting rare jewelry, there is now a catalyst dropdown and quality slider
 - The new mod tags are used to determine if a catalyst affects a mod
 - The base implicit mod is also affected, but only if the item is newly crafted after this commit.
 - Also affects crafting bench and essence mods

NickRyder commented:
Can we split this into 2 PRs, one which just changes the exporter/exported data, and another which does all the actual code changes?

dclamage commented:
This PR now just has the code changes for catalysts

dclamage commented:
Fixed an issue where catalyzed mod lines without ranges would increase exponentially as the item was edited.

Added support for catalyzing uniques
 - Thanks to @LocalIdentity for tagging all the relevant uniques!
 - A jewelry unique gets the catalyst dropdown if any of its mods has any tag
